.circleci/config.yml

```
@@ -429,8 +429,85 @@ jobs:
       ENVOY_VERSIONS: "1.10.0"
     steps: *ENVOY_INTEGRATION_TEST_STEPS
 
+  # This job merges master into the 'current' branch (${CIRCLE_BRANCH}) specified
+  # in the branch filtering of the workflow
+  merge-master:
+    docker:
+      - image: *GOLANG_IMAGE
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+          - c6:96:98:82:dc:04:6c:39:dd:ac:83:05:e3:15:1c:98
+      - checkout
+      - run:
+          name: Merge Consul OSS master branch into current branch
+          command: |
+            set -eu -o pipefail
+
+            # Configure Git
+            git config --global user.email "hashicorp-ci@users.noreply.github.com"
+            git config --global user.name "hashicorp-ci"
+
+            # Fetch latest master
+            git fetch origin
+
+            # Create a merge branch to run tests on
+            git_merge_branch="ci/master-merge-$(date +%Y%m%d%H%M%S)"
+            git checkout -b "${git_merge_branch}"
+            latest_oss_commit="$(git rev-parse origin/master)"
+            git merge -m "Merge Consul OSS branch 'master' at commit ${latest_oss_commit}" origin/master
```

```suggestion
            git merge -m "Merge Consul OSS branch 'master' at commit ${latest_oss_commit}" "${latest_oss_commit}"
```

That way you don't merge commits that happen to happen in between this and the previous statement. Very hypothetical.

---

resolved in the following commit